### PR TITLE
Add FASTQ validation to ingest validation tests

### PR DESCRIPTION
**Pull request by Daghis**
Inform future developers that working with this repository as standalone will not work and provide a reference to the parent repository.

### CODE REVIEW

**Comment by Daghis** [comment]
FYI, I think I've gotten the last few issues resolved since it's running seemingly successfully (if slowly) on `hivevm193`.

**Review by jswelling — CHANGES_REQUESTED** [review]
I found an error while testing in situ- the bad fastq.gz file was identified, but the fact that an error had occurred was lost before the end of the validation run.  I made the following code changes to add diagnostics:
```
diff --git a/src/ingest_validation_tests/fastq_validator.py b/src/ingest_validation_tests/fastq_validator.py
index 848caf9..004a17a 100644
--- a/src/ingest_validation_tests/fastq_validator.py
+++ b/src/ingest_validation_tests/fastq_validator.py
@@ -9,6 +9,9 @@ class FASTQValidator(Validator):
     cost = 5.0
 
     def collect_errors(self, **kwargs) -> List[str]:
-        validator = FASTQValidatorLogic()
+        print('PLUGIN point 1')
+        validator = FASTQValidatorLogic(verbose=True)
+        print('PLUGIN point 2')
         validator.validate_fastq_files_in_path(self.path)
+        print(f'PLUGIN point 3: {validator.errors}')
         return validator.errors
diff --git a/src/ingest_validation_tests/fastq_validator_logic.py b/src/ingest_validation_tests/fastq_validat
index cc6b72d..063d65c 100644
--- a/src/ingest_validation_tests/fastq_validator_logic.py
+++ b/src/ingest_validation_tests/fastq_validator_logic.py
@@ -212,9 +212,11 @@ class FASTQValidatorLogic:
 
             new_errors = self.validate_fastq_file(fastq_file)
             if new_errors:
+                print(f'NEW_ERRORS {new_errors}')
                 errors.extend(new_errors)
             else:
                 # If we successfully process any input file, this can be set.
+                print(f'NO NEW ERRORS')
                 found_one = True
 
         if not found_one:
```

and got the following:

```
[2022-06-03 18:07:16,336] {logging_mixin.py:120} INFO - PLUGIN point 1
[2022-06-03 18:07:16,337] {logging_mixin.py:120} INFO - PLUGIN point 2
[2022-06-03 18:07:16,338] {logging_mixin.py:120} INFO - Validating B001-A-001_S9_L001_R1_001.fastq.gz...
[2022-06-03 19:02:03,244] {logging_mixin.py:120} INFO - NO NEW ERRORS
[2022-06-03 19:02:03,245] {logging_mixin.py:120} INFO - Validating B001-A-001_S9_L001_R2_001.fastq.gz...
[2022-06-03 19:02:03,440] {logging_mixin.py:120} INFO - B001-A-001_S9_L001_R2_001.fastq.gz:137: Line does not begin with '@'.
[2022-06-03 19:02:03,441] {logging_mixin.py:120} INFO - B001-A-001_S9_L001_R2_001.fastq.gz:159: Line does not begin with '+'.
[2022-06-03 19:02:03,444] {logging_mixin.py:120} INFO - B001-A-001_S9_L001_R2_001.fastq.gz:412: Line contains 94 characters which does not match line 410's 91 characters.
[2022-06-03 20:00:29,315] {logging_mixin.py:120} INFO - B001-A-001_S9_L001_R2_001.fastq.gz (421278352 lines) does not match length of B001-A-001_S9_L001_R1_001.fastq.gz (421278356 lines).
[2022-06-03 20:00:29,315] {logging_mixin.py:120} INFO - NEW_ERRORS ["B001-A-001_S9_L001_R2_001.fastq.gz:137: Line does not begin with '@'.", "B001-A-001_S9_L001_R2_001.fastq.gz:159: Line does not begin with '+'.", "B001-A-001_S9_L001_R2_001.fastq.gz:412: Line contains 94 characters which does not match line 410's 91 characters.", 'B001-A-001_S9_L001_R2_001.fastq.gz (421278352 lines) does not match length of B001-A-001_S9_L001_R1_001.fastq.gz (421278356 lines).']
[2022-06-03 20:00:29,315] {logging_mixin.py:120} INFO - Validating B001-A-001_S9_L001_I1_001.fastq.gz...
[2022-06-03 20:23:29,101] {logging_mixin.py:120} INFO - NO NEW ERRORS
[2022-06-03 20:23:29,101] {logging_mixin.py:120} INFO - Validating B001-A-001_S9_L002_R2_001.fastq.gz...
[2022-06-03 21:23:37,603] {logging_mixin.py:120} INFO - NO NEW ERRORS
[2022-06-03 21:23:37,604] {logging_mixin.py:120} INFO - Validating B001-A-001_S9_L002_R1_001.fastq.gz...
[2022-06-03 21:56:08,851] {logging_mixin.py:120} INFO - NO NEW ERRORS
[2022-06-03 21:56:08,934] {logging_mixin.py:120} INFO - Validating B001-A-001_S9_L001_R2_001.fastq.good.gz...
[2022-06-03 21:56:08,934] {logging_mixin.py:120} INFO - NEW_ERRORS ['Filename does not have proper format and will not be processed']
[2022-06-03 21:56:08,949] {logging_mixin.py:120} INFO - PLUGIN point 3: []
```

**Comment by Daghis** [comment]
If the issue is merely that the error messages being generated should be `print()`'d as well as collected in the `List[str]` that's returned, I'd prefer to address that by removing the `verbose` argument and just having all messages be collected *and* printed.

How does that sound?

**Comment by jswelling** [comment]
I've run a test with the fix to errors vs self.errors, and that did indeed fix the problem from the earlier test.  I'm running another test to check on a hint of a second problem- I think an error string from verify_fastq_filename should produce a log message, but should not cause the dataset to be invalid.  Ideally the regex that chooses what to scan could be less broad, and explicitly select *.fastq, *.fastq.gz, (and maybe *.FASTQ.gz; I have seen those around), so that files like fastq.good.gz are never included in the scan in the first place.  I'm thinking I need to do a PR against the test that scans generic *.gz files, so that those scanned by this test aren't scanned again.  One good way to do that mod would be to import the regex used by this test and exclude anything matching that regex.

**Comment by Daghis** [comment]
Comment from @jswelling in Slack:

> here's another delta for the FASTQ validator: when it doesn't find any FASTQ files to check, it needs to not set any errors.  That way non-sequence datasets just skip this validation.  I'll add that to the PR next time I touch it but I thought I'd let you know now.  This was found by accident during David Betancur's experiments with running ingestions on TEST.

The most recent commit logs the no-files-found message, but doesn't return it as an error.

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 xmlschema>=1.6
 tifffile>=2020.10.1
 jsonschema==4.4.0
+git+https://github.com/hubmapconsortium/fastq-utils.git@main#egg=hubmap-fastq-utils
```
Comment (mruffalo) [inline requirements.txt:4]: I recommend changing this to
```
git+https://github.com/hubmapconsortium/fastq-utils@v0.2.5
```
which contains the fix for https://github.com/hubmapconsortium/fastq-utils/issues/1.

Comment (jswelling) [reply requirements.txt]: Done.

**Review by jswelling — APPROVED** [review]
Passed positive, negative, and no-relevant-input test cases in situ on TEST

**Pull request merged**